### PR TITLE
feat: App Info Route

### PR DESCRIPTION
**Pull request by Septias**
To get rid of the big updating button and to make room for more metadata, this PR introduces a dedicated page for information. At the moment only the version and last update are shown, but later we can add stuff like `storage usage` and such.

This PR adds @solidjs/router to the project as two pages are needed. It also moves some state from the store component into a provided context which is usable by both pages (info and store). This adds some complication but was needed IMO. If there is a better option let me know...

I also made some styling changes and added a .editorconfig to configure my editor to use 2 spaces ident in all .tsx which I noticed during development.

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -1,253 +1,66 @@
...
-
-interface AppListProps {
-  items: AppInfoWithState[]
-  search_query: string
-  onDownload: (id: string) => void
-  onForward: (id: string) => void
-  onRemove: (id: string) => void
+import type { StoreRequest } from '~/bindings/StoreRequest'
+
+const metaData = {
```
Comment (Septias) [inline frontend/src/index.tsx:12]: naming this `metaData` feels kind of awkward, maybe @hpk42 has a better idea?

File: frontend/src/store-logic.ts [hunk 2]
```diff
@@ -84,9 +82,7 @@ export async function updateHandler(
       await db.updateMultiple(updated.map(key => ({ ...app_infos[key], state: appInfo[key].state !== AppState.Initial ? AppState.Updating : AppState.Initial })))
     }
 
-    setlastUpdateSerial(payload.serial)
-    setIsUpdating(false)
-    setlastUpdate(new Date())
+    updateDone(payload.serial)
```
Comment (Septias) [inline frontend/src/store-logic.ts:85]: @hpk42 commented this line a while back which is now a singular statement :)

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -11,6 +12,9 @@
     "types": [
       "vite/client"
     ],
+    "paths": {
+      "~/*": ["src/*"]
+    },
```
Comment (Septias) [inline frontend/tsconfig.json:17]: A path alias which allows to no longer us relative paths which will break if you move the file around.

**Comment by hpk42** [comment]
for me starting store.xdc results in blank secreen and the attached errors 
image

File: frontend/src/index.tsx (hunk 1 above)
Comment (hpk42) [reply frontend/src/index.tsx:12]: it's fine for now i think.

**Comment by Septias** [comment]
I have the follow-up pr (where new fields size & date are introduced) ready so would be nice to get this in soon

**Review by hpk42 — APPROVED** [review]
said in chat already it's fine to merge -- note that when you change the store.xdc you should increment the version each time.

**Comment by Septias** [comment]
changing the implementation to a simpler one atm, so it doesn't even compile now. Will fix it tomorrow

**Review by hpk42 — APPROVED** [review]
works well.

**Pull request merged**